api/zulip/zulip_bot_server.py

```diff
@@ -0,0 +1,99 @@
+import json
+from typing import Any, Dict, Mapping, Union, List
+from werkzeug.exceptions import BadRequest
+from six.moves.configparser import SafeConfigParser
+import optparse
+from importlib import import_module
+
+from zulip import Client
+from bots_api.bot_lib import BotHandlerApi, StateHandler
+
+bots_config = {}  # type: Dict[str, Mapping[str, str]]
+available_bots = []  # type: List[str]
+bots_lib_module = {}  # type: Dict[str, Any]
+
+def read_config_file(config_file_path):
+    # type: (str) -> None
+    parser = SafeConfigParser()
+    parser.read(config_file_path)
+
+    for section in parser.sections():
+        bots_config[section] = {
+            "email": parser.get(section, 'email'),
+            "key": parser.get(section, 'key'),
+            "site": parser.get(section, 'site'),
+        }
+
+def load_lib_modules():
+    # type: () -> None
+    for bot in available_bots:
+        try:
+            package = 'bots.{}'.format(bot)
+            bots_lib_module[bot] = import_module('.' + bot, package=package)
+        except ImportError:
+            print("\n Import Error: Bot \"{}\" doesn't exists. Please make sure you have set up the flaskbotrc "
+                  "file correctly.\n".format(bot))
+            sys.exit(1)
+
+app = Flask(__name__)
+
+@app.route('/bots/<bot>', methods=['POST'])
+def handle_bot(bot):
+    # type: (str) -> Union[str, BadRequest]
+    if bot not in available_bots:
+        return BadRequest("requested bot service {} not supported".format(bot))
+
+    client = Client(email=bots_config[bot]["email"],
+                    api_key=bots_config[bot]["key"],
+                    site=bots_config[bot]["site"])
+    try:
+        restricted_client = BotHandlerApi(client)
+    except SystemExit:
+        return BadRequest("Cannot fetch user profile for bot {}, make sure you have set up the flaskbotrc "
+                          "file correctly.".format(bot))
+    message_handler = bots_lib_module[bot].handler_class()
+
+    # TODO: Handle stateful bots properly.
+    state_handler = StateHandler()
+
+    event = json.loads(request.data)
+    message_handler.handle_message(message=event["message"],
+                                   bot_handler=restricted_client,
+                                   state_handler=state_handler)
+    return "Success!"
+
+def main():
+    # type: () -> None
+    usage = '''
+                zulip-bot-server <path to flaskbotrc>
+                Example: zulip-bot-server ~/flaskbotrc
+                (This program loads the bot configurations from the
+                config file (flaskbotrc here) and loads the bot modules.
+                It then starts the server and fetches the requests to the
+                above loaded modules and returns the success/failure result)
+                Please make sure you have a current flaskbotrc file with the
+                configurations of the required bots.
+                See lib/readme.md for more context.
+                '''
```

Comment:
This description could be improved a bit. For now, get rid of the parenthesis (`(This program ... result)`). And instead of `from the config file (flaskbotrc here)`, replace it with `from the flaskbotrc config file`. :)